### PR TITLE
@types/htmlparser2 Make DOMHandler events optional

### PR DESCRIPTION
**Pull request by paambaati**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/fb55/htmlparser2/wiki/Parser-options#events
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by paambaati** [comment]
This PR makes all events optional on the DOMHandler class, so it can be used in htmlparser2 with only the event listeners we need. Without these being optional, we're forced to implement every listener (we don't have to).

**Comment by paambaati** [comment]
CI should pass once #36588 lands.

**Comment by paambaati** [comment]
@typescript-bot I'm waiting on #36588 to be merged.

CC: @RyanCavanaugh

File: types/htmlparser2/index.d.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@
 // Definitions by: James Roland Cabresos <https://github.com/staticfunction>
 //                 Linus Unnebäck <https://github.com/LinusU>
 //                 Johan Davidsson <https://github.com/johandavidson>
+//                 GP <https://github.com/paambaati>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
```
Comment (WillGibson) [inline types/htmlparser2/index.d.ts]: I'm not convinced we should be locking this to TypeScript 2.2, e.g I know at least one place it's being used with TypeScript 3.x.

**Review by WillGibson — COMMENTED** [review]
One comment. Plus there is a (possibly unrelated) pipeline fail which make me nervoous.

File: types/htmlparser2/index.d.ts (hunk 1 above)
Comment (paambaati) [reply types/htmlparser2/index.d.ts]: @WillGibson This is the minimum version of TS that these typings are guaranteed to work with. It works fine on TS 3.x.

Comment (WillGibson) [reply types/htmlparser2/index.d.ts]: Ah, cool.

**Pull request closed without merging**